contrib/win32/win32compat/fileio.c

```diff
@@ -1106,39 +1044,159 @@ fileio_readlink(const char *path, char *buf, size_t bufsiz)
 		goto cleanup;
 	}
 
-	/* read the link data from the passed symlink */
-	resolved_link = fileio_readlink_internal(wpath);
-	if (resolved_link == NULL) {
-		/* errorno will have been set by called function */
+	/* obtain a handle to send to deviceioctl  */
+	handle = CreateFileW(wpath, 0, 0, NULL, OPEN_EXISTING, 
```

Comment:
wpath is NULL here.
I think you forgot to initialize wpath to have value of path?